cfgov/cfgov/urls.py

```diff
@@ -413,53 +415,98 @@
-        url(r'^(?i)ask-cfpb/api/autocomplete/$',
-            ask_autocomplete, name='ask-autocomplete-en'),
-        url(r'^(?P<language>es)/obtener-respuestas/api/autocomplete/$',
-            ask_autocomplete, name='ask-autocomplete-es'),
+
+# New Spanish Ask CFPB URLs
+# These are defined here so they can be include()ed by the feature flag below
+ask_cfpb_spanish_patterns = [
+    url(r'^c/(.+)/(?P<ask_id>\d+)/(.+)\.html$',
+        RedirectView.as_view(
+            url='/es/obtener-respuestas/slug-es-%(ask_id)s',
+            permanent=True)),
+    url(r'^([-\w]{1,244})-(es)-(\d{1,6})/?$',
+        view_answer,
+        name='ask-spanish-answer'),
+    url(r'^([-\w]{1,244})-(es)-(\d{1,6})/imprimir/?$',  # noqa: E501
+        print_answer,
+        name='ask-spanish-print-answer'),
+    url(r'^buscar/$',
+        ask_search,
+        kwargs={'language': 'es'},
+        name='ask-search-es'),
+    url(r'^buscar/(?P<as_json>json)/$',  # noqa: E501
+        ask_search,
+        kwargs={'language': 'es'},
+        name='ask-search-es-json'),
+    url(r'^api/autocomplete/$',
+        ask_autocomplete,
+        kwargs={'language': 'es'},
+        name='ask-autocomplete-es'),
+
+    # Answer /es/obtener-respuestas landing page with Wagtail.
+    # This is necessary because the pattern is shared with the old
+    # knowledgebase urls and we have to explicitly offer it as an alternative
+    # in the flagged_url() for /es/obtener-respuestas below.
+    # Once the feature flag is removed this will be unnecessary.
+    url(r'^$',
+        lambda req: ServeView.as_view()(req, req.path)),
+]
+
+# New English Ask CFPB URL redirects
+# These are defined here so they can be include()ed by the feature flag below.
+ask_cfpb_english_redirects = [
+    url(r'^$',
+        RedirectView.as_view(
+            url='/ask-cfpb/',
+            permanent=True)),
+    url(r'^(?P<ask_id>\d+)/(.*)$',
+        RedirectView.as_view(
+            url='/ask-cfpb/slug-en-%(ask_id)s',
+            permanent=True)),
+    url(r'^search/\?selected_facets=category_exact:(?P<category>[^&]+)',  # noqa: E501
+        RedirectView.as_view(
+            url='/ask-cfpb/category-%(category)s',
+            permanent=True)),
+]
+
+# Flagged includes for either old knowledgebase Ask CFPB URLs or new ask_cfpb
+# URLs. In this way the shared URLs askcfpb/ and es/obtener-respuestas/ will
+# either serve the new ask_cfpb or will fallback on the old knowledgebase.
+flagged_kb_patterns = [
+    flagged_url('WAGTAIL_ASK_CFPB',
+                r'^(?i)askcfpb/',
+                include(ask_cfpb_english_redirects),
+                fallback=include_if_app_enabled(
+                    'knowledgebase', 'knowledgebase.urls')),
+    flagged_url('WAGTAIL_ASK_CFPB',
+                r'^es/obtener-respuestas/',
+                include(ask_cfpb_spanish_patterns),
+                fallback=include_if_app_enabled(
+                    'knowledgebase', 'knowledgebase.babel_urls')),
+]
```

Comment:
So, this is a fairly big change, and it breaks up the new ask URLs but I think it's the only way we can get a clean knowledgebase/ask_cfpb transition with the feature flag for URLs they share. We'll group them together once we retire the feature flag after launch.

What's going on here is this: cfpb/wagtail-flags#13 introduces fallback `include()`s, which means for `askcfpb/` and `es/obtener-respuestas/` we can include our new URLs if the flag is enabled, or fallback on the old knowledgebase URLs if it's disabled. I've put the English and Spanish URLs *that are shared* (the english URLs are just redirects) in separate lists above. This lets us `include()` them and fallback on the `include_if_app_enabled()` for the knowledgebase URLs. The new URLs that we're introducing are below and simply enabled with the `WAGTAIL_ASK_CFPB` flag (that hasn't changed).

For this to work right now (until a new release of Wagtail-Flags, which I'll make as soon as I know this works), you'll have to install it in your cfgov-refresh virtualenv manually:

```
pip install git+https://github.com/cfpb/wagtail-flags.git@fallback-include
```

@higs4281 if you have a chance, it'd be awesome if you could make sure this makes sense.

@virginiacc could you confirm that the URLs that should work do work? Since the redirects are permanent, you'll probably want to use incognito mode to try them, otherwise Chrome will cache them until browsing data is cleared (yes, that is the voice of experience 😉 ).